React/CoreModules/RCTDevSettings.mm

```diff
@@ -119,7 +119,11 @@ - (void)_reloadWithDefaults:(NSDictionary *)defaultValues
       _settings[key] = defaultValues[key];
     }
   }
-  [_userDefaults setObject:_settings forKey:kRCTDevSettingsUserDefaultsKey];
+  
+  // TODO(macOS ISS#2323203): protect against race conditions where another thread holds a mutext trying to set this at the same time
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [self->_userDefaults setObject:self->_settings forKey:kRCTDevSettingsUserDefaultsKey];
+  });
```

Comment:
Could this cause an issue because this method is called on init, and someone might potentially use this object before all the default values are set because it's asynchronous?

Reply:
nit: maybe we should only dispatch async to main thread when the current running thread isn't main thread.... or do you know already it will always be in background thread?

Reply:
There's a utility in React Native that does this exact check too (https://github.com/facebook/react-native/blob/1163599989aa6d641f127f3100a01c45f04d2edc/React/Base/RCTUtils.m#L259)

Reply:
Note that the utility next to it that does a `dispatch_sync` is explicitly named unsafe because of deadlocks exactly like this one :D

Reply:
Nice yea I think it makes sense to use RN's explicit method to check the thread. I don't think we can guarantee this is ever called on a given thread

Reply:
> Could this cause an issue because this method is called on init, and someone might potentially use this object before all the default values are set because it's asynchronous?

Don't think we need to worry about that. The `_settings` var is determined synchronously before we call this and the `_userDefaults` value is set up synchronously before this in init.